projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -323,7 +340,14 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
                 const input = this.igxForOf[i];
                 const embeddedView = this.dc.instance._vcr.createEmbeddedView(
                     this._template,
-                    { $implicit: input, index: this.igxForOf.indexOf(input) }
+                    { 
+                        $implicit: input, 
+                        index: this.igxForOf.indexOf(input), 
+                        first: this.first(this.igxForOf.indexOf(input)),
```

Comment:
don't indexOf for each asignment, also this should just be *i* (not sure why it was indexOf(input) before but just try with i and see if the tests succeed

Reply:
Done